### PR TITLE
Added clear all button for mobile and tablet

### LINKED ISSUE #401: Clear all button

### PR DESCRIPTION
**Pull request by rjmcneill**
Closes #401 

The spacing isn't right with the period selectors at the minute, but @jdhodges431 is working on those at the minute.

@tlsim I had to rebase your stuff into this one, let me know if I've missed anything :smile:

### CODE REVIEW

File: src/assets/js/app.js [hunk 1]
```diff
@@ -262,8 +262,21 @@ export default function() {
                 updateModelSelectedPeriod(period.option);
                 _dataInterface(period.option.seconds);
                 render();
+            })
+            .on(event.clearAll, function() {
+                for (var i = 0; i < primaryChartModel.indicators.length; i++) {
```
Comment (jleft) [inline src/assets/js/app.js]: Consider using a `array.forEach`

Comment (jleft) [inline src/assets/js/app.js]:
```js
function deselect(option) { option.isSelected = false; }

primaryChartModel.indicators.forEach(deselect);
```

File: src/assets/js/app.js [hunk 2]
```diff
@@ -280,28 +293,37 @@ export default function() {
...
-                });
-                // TODO: Remove .remove! (could a secondary chart group component manage this?).
-                containers.secondaries.selectAll('*').remove();
-                updateLayout();
-                render();
+                renderSecondaryCharts();
+            });
+    }
+
+    function renderPrimaryIndicators() {
```
Comment (jleft) [inline src/assets/js/app.js]: I find these function names misleading as they render everything, not just the primary indicators/secondary charts. Something like `updatePrimaryChartIndicatorsAndRender()` could be okay, however...

It also means on the `clearAll` event we'll render twice, when we only need to render once?

I'd consider changing these to just manipulate the state

File: src/assets/js/event.js [hunk 3]
```diff
@@ -8,6 +8,7 @@ export default {
     dataProductChange: 'dataProductChange',
     dataPeriodChange: 'dataPeriodChange',
     resetToLatest: 'resetToLatest',
+    clearAll: 'clearAll',
```
Comment (jleft) [inline src/assets/js/event.js]: I'd consider renaming this to something more specific - maybe `clearIndicators` or `clearAllIndicators`?

File: src/assets/js/menu/head.js [hunk 4]
```diff
@@ -42,6 +44,11 @@ export default function() {
                 })
                 .call(dataPeriodSelector);
 
+            container.select('#clear-indicators')
+                .on('click', function() {
```
Comment (jleft) [inline src/assets/js/menu/head.js]: Can do this, which is a bit neater:

```js
.on('click', dispatch[event.clearAll]);
```

File: src/assets/styles/head.less [hunk 5]
```diff
@@ -44,5 +44,9 @@
         display: block;
       }
     }
+
+    .delete-button {
+      margin: 16px 4px 16px auto;
```
Comment (jleft) [inline src/assets/styles/head.less]: Where do these numbers come from? Can they be derived from a variable(s)? Would it be suitable to use flex for the positioning?

**Comment by jleft** [comment]
:+1: Happy to merge this once rebased with develop and integrated with the changes in #465

**Comment by rjmcneill** [comment]
@jleft  I'll get that sorted today :smile:

File: src/assets/styles/head.less [hunk 6]
```diff
@@ -55,6 +55,11 @@
 
     #mobile-period-selector {
       align-self: center;
+      margin-right: 12px;
+    }
+
+    .delete-button {
+      margin: auto 12px auto auto;    // These may need changed when #465 goes in
```
Comment (jleft) [inline src/assets/styles/head.less]: Can we make use `align-self: center;`? Similar to `#mobile-period-selector`?

File: src/index.html.tpl [hunk 7]
```diff
@@ -16,6 +16,9 @@
             <div id="product-dropdown" class="dropdown product-dropdown"></div>
             <div id="period-selector" class="hidden-xs hidden-sm"></div>
             <div id="mobile-period-selector" class="hidden-md hidden-lg dropdown"></div>
+            <div id="period-selector" class="hidden-xs hidden-sm"></div>
```
Comment (jleft) [inline src/index.html.tpl]: This is a duplicate of line 17 - remove.

File: src/index.html.tpl [hunk 8]
```diff
@@ -16,6 +16,9 @@
...
+            <div id="mobile-period-selector" class="hidden-md hidden-lg dropdown"></div>
```
Comment (jleft) [inline src/index.html.tpl]: This is a duplicate of line 18 - remove.

**Comment by rjmcneill** [comment]
Thanks @jleft !

**Comment by jleft** [comment]
Thank you!

**Pull request merged**